spec/features/create_and_deliver_digests_spec.rb

```diff
@@ -217,4 +217,208 @@ def second_expected_daily_email_body(subscription)
+
+      Change note two: Description two
+
+      Updated at 09:00 am on 27 December 2016
+
+      ---
+
+      Unsubscribe from [Subscriber list one](http://www.dev.gov.uk/email/unsubscribe/#{subscription_one.uuid}?title=Subscriber%20list%20one)
+
+      &nbsp;
+
+      #Subscriber list two
+
+      [Title four](http://www.dev.gov.uk/base-path)
+
+      Change note four: Description four
+
+      Updated at 09:30 am on 1 January 2017
+
+      ---
+
+      [Title three](http://www.dev.gov.uk/base-path)
+
+      Change note three: Description three
+
+      Updated at 09:00 am on 30 December 2016
+
+      ---
+
+      Unsubscribe from [Subscriber list two](http://www.dev.gov.uk/email/unsubscribe/#{subscription_two.uuid}?title=Subscriber%20list%20two)
+    BODY
+  end
+
+  def second_expected_weekly_email_body(subscription)
+    <<~BODY
+      #Subscriber list one
+
+      [Title one](http://www.dev.gov.uk/base-path)
+
+      Change note one: Description one
+
+      Updated at 10:00 am on 28 December 2016
+
+      ---
+
+      [Title two](http://www.dev.gov.uk/base-path)
+
+      Change note two: Description two
+
+      Updated at 09:00 am on 27 December 2016
+
+      ---
+
+      Unsubscribe from [Subscriber list one](http://www.dev.gov.uk/email/unsubscribe/#{subscription.uuid}?title=Subscriber%20list%20one)
+    BODY
+  end
+  scenario "weekly digest run" do
+    login_with_internal_app
+
+    #create two subscriber lists with different links
+    list_one_topic_id = "0eb5d0f0-d384-4f27-9da8-3f9e9b22a820"
+    list_two_topic_id = "a915e039-070b-4633-813d-187af61cad7a"
+
+    stub_govdelivery("TEST123")
+    subscribable_one_id = create_subscribable(title: "Subscriber list one", links: {
+      topics: [list_one_topic_id]
+    })
+
+    stub_govdelivery("TEST456")
+    subscribable_two_id = create_subscribable(title: "Subscriber list two", links: {
+      topics: [list_two_topic_id]
+    })
+
+    #create two daily subscribers, one subscribed to daily digests for both
+    #subscribables and the other for daily for subscribable_one only
+    subscriber_one_address = "test-one@example.com"
+    subscriber_two_address = "test-two@example.com"
+
+    non_weekly_digest_subscriber_address = "test-three@example.com"
+
+    subscribe_to_subscribable(
+      subscribable_one_id,
+      address: subscriber_one_address,
+      frequency: Frequency::WEEKLY
+    )
+
+    subscribe_to_subscribable(
+      subscribable_two_id,
+      address: subscriber_one_address,
+      frequency: Frequency::WEEKLY
+    )
+
+    subscribe_to_subscribable(
+      subscribable_one_id,
+      address: subscriber_two_address,
+      frequency: Frequency::WEEKLY
+    )
+
+    subscribe_to_subscribable(
+      subscribable_two_id,
+      address: non_weekly_digest_subscriber_address,
+      frequency: Frequency::DAILY
+    )
+
+    #publish two items to each list
+    Timecop.freeze "2016-12-28 10:00" do
+      create_content_change(
```

Comment:
I wonder if we could make these into methods used across the daily and weekly digests? `create_content_change_1`?

I like that they're separate in each test right now, but it's also a lot of code duplication. What do you think?

Reply:
I don't mind duplication within the tests. The other issue is that the dates are different in the daily and the weekly tests so we'd have to pass them in if we extract a method. Happy to do that but I prefer it as it is.

Reply:
Yeah, that's fine with me. I agree the code duplication is not too much of an issue.